test/utils/gtest_formating_checks.py

```diff
@@ -0,0 +1,89 @@
+import os
+import re
+import logging
+
+logging.basicConfig(level=logging.INFO, format='%(levelname)s: %(message)s')
+
+FOLDER_PATH = "../../test/gtest"
+
+# Ignore list: Add test names or file paths you want to exclude
+IGNORE_LIST = {
+    "CPU_MIOpenDriverRegressionBigTensorTest_FP32",
+    "../../test/gtest/reduce_custom_fp32.cpp",
+}
+
+# Valid enums and Regex for validation
+VALID_HW_TYPES = {"CPU", "GPU"}
+VALID_DATATYPES = {"FP8", "FP16", "FP32", "FP64", "BFP16", "BFP8", "I64", "I32", "I16", "I8", "NONE"}
+TESTSUITE_REGEX = re.compile(
+    r"^(CPU|GPU)_[A-Za-z0-9]+(?:_[A-Za-z0-9]+)*_(" + "|".join(VALID_DATATYPES) + r")$"
+)
+TEST_P_REGEX = re.compile(r"TEST_P\(([^,]+),\s*([^)]+)\)")
+INSTANTIATE_TEST_REGEX = re.compile(r"INSTANTIATE_TEST_SUITE_P\(\s*([^\n,]+),\s*([^\n,]+),")
+TEST_TYPE_REGEX = re.compile(r"^(Smoke|Full|Perf|Unit)([A-Za-z0-9]*)?$")
+
+
+def analyze_tests(folder_path):
+    errors = []
+
+    for root, _, files in os.walk(folder_path):
+        for file in files:
+            if file.endswith(".cpp"):
+                file_path = os.path.join(root, file)
+
+                if file_path in IGNORE_LIST:
+                    logging.info(f"Skipping ignored file: {file_path}")
+                    continue
+
+                with open(file_path, "r") as f:
+                    content = f.read()
+
+                # Extract TEST_P and INSTANTIATE_TEST_SUITE_P
+                test_p_matches = TEST_P_REGEX.findall(content)
+                instantiate_matches = INSTANTIATE_TEST_REGEX.findall(content)
+
+                test_p_suites = {suite: info for suite, info in test_p_matches}
+                instantiated_suites = {suite: test_type for test_type, suite in instantiate_matches}
+
+                # Validate TEST_P suites
+                for suite, info in test_p_suites.items():
+                    if suite in IGNORE_LIST:
+                        logging.info(f"Skipping ignored test suite: {suite}")
+                        continue
+
+                    if not TESTSUITE_REGEX.match(suite):
+                        errors.append(f"{file_path}: Invalid TESTSUITE_NAME '{suite}' in TEST_P.")
+
+                    if suite not in instantiated_suites:
```

Comment:
It's nice to catch these before runtime! It does need to be aware of GTEST_ALLOW_UNINSTANTIATED_PARAMETERIZED_TEST though, and it also gives false failures when macros are used, e.g. layout_transpose.cpp. I don't think we should discard that flexibility.